### PR TITLE
test: fix the problem of unit test

### LINKED ISSUE #3109: Unit Test class TmNettyClientTest.java didn't get expected results
## Why you need it?

Unit test classes didn't get expected results

## How it could be?

update TmNettyClientTest

## Other related information

https://github.com/seata/seata/blob/fc0c9ced86c317fafb2bffd37289d8afa56e7be8/test/src/test/java/io/seata/core/rpc/netty/TmNettyClientTest.java#L71

### PR DESCRIPTION
**Pull request by LiWenGu**
### Ⅰ. Describe what this PR did

fix the problem of unit test

### Ⅱ. Does this pull request fix one issue?

fixes #3109

### CODE REVIEW

**Comment by lightClouds917** [comment]
The CI failed .Please check .
```
[ERROR] Failures: 
[ERROR]   TmNettyClientTest.testSendMsgWithResponse:135 expected: <RuntimeException[SessionManager is NOT init!]> but was: <TransactionException[Could not found global transaction xid = 127.0.0.1:8091:1249853, may be has finished.]>
[INFO] 
[ERROR] Tests run: 84, Failures: 1, Errors: 0, Skipped: 4
```

File: test/src/test/java/io/seata/core/rpc/netty/TmNettyClientTest.java [hunk 1]
```diff
@@ -47,17 +47,20 @@
      */
     @Test
     public void testDoConnect() throws Exception {
-
+        ThreadPoolExecutor workingThreads = initMessageExecutor();
+        NettyRemotingServer nettyRemotingServer = new NettyRemotingServer(workingThreads);
         //start services server first
-        workingThreads.submit(new Runnable() {
```
Comment (LiWenGu) [inline test/src/test/java/io/seata/core/rpc/netty/TmNettyClientTest.java:52]: exeucte will swallow the exception

**Comment by a364176773** [comment]
mvn clean install -DskipTests=false -Dcheckstyle.skip=false -Dlicense.skip=false

please  use this command to test locally

File: test/src/test/java/io/seata/saga/engine/db/StateMachineDBTests.java [hunk 2]
```diff
@@ -27,10 +27,7 @@
 import io.seata.saga.statelang.domain.StateMachineInstance;
 import io.seata.tm.api.GlobalTransaction;
 import io.seata.tm.api.GlobalTransactionContext;
-import org.junit.jupiter.api.Assertions;
-import org.junit.jupiter.api.BeforeAll;
-import org.junit.jupiter.api.Disabled;
-import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.*;
```
Comment (slievrly) [inline test/src/test/java/io/seata/saga/engine/db/StateMachineDBTests.java:30]: don't use `*`

**Comment by a364176773** [comment]
[ERROR]   TmNettyClientTest.testSendMsgWithResponse:148 expected: <RuntimeException[SessionManager is NOT init!]> but was: <TransactionException[Could not found global transaction xid = 127.0.0.1:8091:1249853, may be has finished.]>

**Comment by slievrly** [comment]
@LiWenGu 
image

**Comment by LiWenGu** [comment]
> @LiWenGu
> image

https://github.com/seata/seata/blob/772d55d912ffd03c0c8feb5e1670144b5530636c/server/src/main/java/io/seata/server/coordinator/AbstractCore.java#L115-L123
这个单元测试应该在117行就报错的，在119行报错，说明有其它的干扰，即 SessionHolder 已被 Thread 绑定。还有一个是错误日志还有个报错，是 `java.net.BindException: Address already in use`，说明这个单元测试的客户端请求是某个不知名的单元测试的服务端（服务端没有正确关闭）。所以我猜测有两个问题：
1. SessionHolder 被干扰
2. 服务端启动被干扰

**Comment by a364176773** [comment]
请解决代码冲突

**Review by slievrly — APPROVED** [review]
LGTM

**Pull request merged**